ProcessSlave.php

```diff
@@ -424,11 +431,13 @@ protected function serveStatic(\React\Http\Request $request, HttpResponse $respo
             $path = str_replace("\\", '/', $path);
         }
 
-        $basePath = realpath($this->getBridge()->getStaticDirectory());
-        $filePath = realpath($basePath . $path);
+        if (!isset($this->staticBasePath)) {
+            $this->staticBasePath = realpath($this->getBridge()->getStaticDirectory());
+        }
+        $filePath = realpath($this->staticBasePath . $path);
 
-        // prevent access outside basePath
-        if (strpos($basePath, $filePath) !== 0) { 
+        // prevent access outside base path
+        if (strpos($this->staticBasePath, $filePath) !== 0) { 
```

Comment:
The two arguments of strpos are swapped:

If for example `$this->staticBasePath` is `/foo` and `$path` is `bar`, `$filePath` will be `/foo/bar`. In this case the _haystack_ argument must be `$filePath` and therefore `/foo/bar` and not `/foo`.